ProcessLib/HeatTransportBHE/HeatTransportBHEProcess.cpp

```diff
@@ -239,18 +277,37 @@ void HeatTransportBHEProcess::createBHEBoundaryConditionTopBottom(
             for (auto const& in_out_component_id :
                  bhe.inflow_outflow_bc_component_ids)
             {
-                // Top, inflow.
-                bcs.addBoundaryCondition(
-                    ProcessLib::createBHEInflowDirichletBoundaryCondition(
-                        get_global_bhe_bc_indices(bc_top_node_id,
-                                                  in_out_component_id),
-                        bhe));
-
-                // Bottom, outflow.
-                bcs.addBoundaryCondition(
-                    ProcessLib::createBHEBottomDirichletBoundaryCondition(
-                        get_global_bhe_bc_indices(bc_bottom_node_id,
-                                                  in_out_component_id)));
+                if (bhe.bhe_if_use_python_bc == true)
+                // call BHEPythonBoundarycondition
+                {
+                    // Top, inflow.
+                    bcs.addBoundaryCondition(
+                        ProcessLib::createBHEInflowPythonBoundaryCondition(
+                            get_global_bhe_bc_indices(bc_top_node_id,
+                                                      in_out_component_id),
+                            _process_data.py_bc_object));
+
+                    // Bottom, outflow.
+                    bcs.addBoundaryCondition(
+                        ProcessLib::createBHEBottomDirichletBoundaryCondition(
+                            get_global_bhe_bc_indices(bc_bottom_node_id,
+                                                      in_out_component_id)));
+                }
+                else
+                {
+                    // Top, inflow.
+                    bcs.addBoundaryCondition(
+                        ProcessLib::createBHEInflowDirichletBoundaryCondition(
+                            get_global_bhe_bc_indices(bc_top_node_id,
+                                                      in_out_component_id),
+                            bhe));
+
+                    // Bottom, outflow.
+                    bcs.addBoundaryCondition(
+                        ProcessLib::createBHEBottomDirichletBoundaryCondition(
```

Comment:
Code duplication for the bottom BC is not needed.

Reply:
fixed in commit #95f4087el